src/pages/es/guides/client-side-scripts.md

```diff
@@ -0,0 +1,195 @@
+
+En su lugar, puedes usar [`addEventListener`](https://developer.mozilla.org/es/docs/Web/API/EventTarget/addEventListener) en una etiqueta `<script>` para manejar interaciones de usuario.
+
+```astro title="src/components/AlertButton.astro"
+<button class="alert">Click me!</button>
+
+<script>
+  // Encuentra todos los botones con la clase `alert` en la página.
+  const buttons = document.querySelectorAll('button.alert');
+
+  // Maneja los clics en cada botón.
+  buttons.forEach((button) => {
+    button.addEventListener('click', () => {
+      alert('Button was clicked!');
+    });
+  });
+</script>
+```
+
+:::note
+Si tienes multiples componentes `<AlertButton />` en una página, Astro no ejecutará el script múltiples veces. Los scripts son empaquetaod y solo incluidos una vez en la página. Usando `querySelectorAll` asegura que este script agregue el event listener a cada boton con la clase `alert` en la página.
+:::
+
+### Componentes Web con elementos personalizados.
+
+Puedes crear tus propios elementos HTML con comportamiento personalizado usando el estándar de componentes Web. Definir un [elemento personalizado](https://developer.mozilla.org/es/docs/Web/Web_Components/Using_custom_elements) en un componente `astro` te permite crear componentes interactivos sin necesidad de un framework.
+
+```astro title="src/components/AstroHeart.astro"
+<!-- Envuelve los elementos del componente en nuestro elemento personalizado “astro-heart”. -->
+<astro-heart>
+  <button aria-label="Heart">💜</button> × <span>0</span>
+</astro-heart>
+
+<script>
+  // Define el comportamiento para nuestro nuevo tipo de elemento HTML.
+  class AstroHeart extends HTMLElement {
+    constructor() {
+			super();
+      let count = 0;
+
+      const heartButton = this.querySelector('button');
+      const countSpan = this.querySelector('span');
+
+      // Cada vez que el button es clickeado, actualiza el contador.
+			heartButton.addEventListener('click', () => {
+        count++;
+        countSpan.textContent = count;
+      });
+		}
+  }
+
+  // Dile al navegador que use nuestra clase AstroHeart para elementos <astro-heart>.
+  customElements.define('astro-heart', AstroHeart);
+</script>
+```
+
+Hay dos ventajas de usar un elemento personalizod aqui:
+
+1. En lugar de buscar en toda la pagina usando `document.querySelector()`, puedes usar `this.querySelector()`, el cual solo busca dentro de la instancia del elemento personalizado. Esto hace más fácil para trabajar  con solo los hijos de una instancia del componente a la vez.
+
+2. Aunque un `<script>` solo se ejecuta una vez, el navegador ejecutará método `constructor()` de nuestro elemento cada vez que encuentre `<astro-heart>` en la página. Esto significa que puedes escribir seguramente código para un componente a la vez, incluso si tienes la intención de utilizar este componente varias veces en una página.
```

Comment:
I found the word "seguramente" a bit confusing in this context. What about something like this:

" 2. Aunque el `<script>` solo se ejecuta una vez, el navegador ejecutará el método `constructor()` de nuestro elemento cada vez que encuentre `<astro-heart>` en la página. Esto significa que puedes escribir código de forma segura para un componente a la vez, incluso si tienes la intención de utilizar este componente varias veces en una página."